Python/Tests/Core.UI/TestExplorerTests.cs

```diff
@@ -122,6 +169,31 @@ public class TestExplorerTests {
             return workspaceFolderPath;
         }
 
+        private static void DebugTest(PythonVisualStudioApp app, TestInfo test, TestInfo[] tests) {
+            var testExplorer = app.OpenTestExplorer();
+            Assert.IsNotNull(testExplorer, "Could not open test explorer");
+
+            Console.WriteLine("Waiting for tests discovery");
+            app.WaitForOutputWindowText("Tests", $"Discovery finished: {tests.Length} tests found", 15_000);
+
+            testExplorer.GroupByProjectNamespaceClass();
+
+            var item = testExplorer.WaitForItem(test.Path);
+            Assert.IsNotNull(item, $"Coult not find {string.Join(":", test.Path)}");
+
+            var breakLineno = test.SourceLine + 1;
+            app.Dte.Debugger.Breakpoints.Add(File: test.SourceFile, Line: breakLineno);
+
+            Console.WriteLine($"Debugging test {test.Name}");
+
+            testExplorer.DebugAll();
+
+            app.WaitForMode(EnvDTE.dbgDebugMode.dbgBreakMode);
+
+            Assert.IsNotNull(app.Dte.Debugger.BreakpointLastHit);
+            Assert.AreEqual(breakLineno, app.Dte.Debugger.BreakpointLastHit.FileLine);
+        }
```

Comment:
The test ends while being stopped at a breakpoint. How does that behave? Say you run all 4 debug ui tests?

Reply:
I think you're right... when I ran these all it didn't have a problem but it might for other tests. i'll see what the regular debug tests do for cleanup

Reply:
updated